main/views.py

```diff
@@ -438,9 +439,190 @@
+#                 "unix_timestamp": payment.unix_timestamp,
+#                 "amount": str(payment.amount),
+#                 "testing": "0" if is_real_payment else "1",
+#                 "description": payment.description,
+#                 "order_id": order_id,
+#                 "client_email": payment.client_email,
+#                 "success_url": payment.success_url,
+#                 "receipt_items": receipt_items,
+#             }
+#
+#             # Use the appropriate secret key based on whether it's a real or test payment
+#             # Bug: using env var for PAYMENT_SECRET_KEY sometimes doesn't work when calculating the signature for real payment
+#             secret_key = settings.PAYMENT_SECRET_KEY if is_real_payment else settings.PAYMENT_TEST_SECRET_KEY
+#             generator = MerchantSignatureGenerator(secret_key)
+#             payment.signature = generator.get_signature(params)
+#             params["signature"] = payment.signature
+#
+#             # Send POST request to Modulbank
+#             response = requests.post("https://pay.modulbank.ru/pay", data=params)
+#             pprint(f"Igor1 {response.url=}")
+#
+#             if response.status_code == 200:
+#                 pprint(f"Igor2 Good {response.status_code=}")
+#                 payment.save()
+#                 return HttpResponse(response.text)
+#             else:
+#                 pprint(f"Igor3 {payment.signature=}\n{response.status_code=}")
+#                 return HttpResponse(response.text)
+#     else:
+#         form = PaymentForm(
+#             initial={
+#                 "amount": plan.price,
+#                 "description": f"Абонентская оплата. Тариф: {plan_name_readable}",
+#                 "unix_timestamp": unix_timestamp,
+#                 "client_email": request.user.tenant,
+#                 "receipt_items": receipt_items,
+#                 # Add other initial values as needed
+#             }
+#         )
+#         pprint("Form not valid")
+#
+#     context = {
+#         "form": form,
+#         "plan": plan,
+#         "is_real_payment": is_real_payment,
+#     }
+#     if request.htmx:
+#         # dialog_html = render_to_string("main/partials/payment_plan_modal.html", context)
+#         dialog_html = render_to_string("main/partials/payment_plan_modal.html", context)
+#         return HttpResponse(dialog_html)
+#         # return render(request, "main/partials/payment_plan_modal.html", context)
+#     return render(request, "main/billing.html", context)
+
+
+def billing_view(request):
+    plan_name = request.GET.get(
+        "plan", str(request.user.tenant.payment_plan.name)
+    )  # Default is the user's current plan
+    plan = get_object_or_404(PaymentPlan, name=plan_name)
+    unix_timestamp = int(timezone.now().timestamp())
+
+    try:
+        order_id = create_unique_order_id(tenant_id=request.user.tenant.id)
+    except ValueError as e:
+        return HttpResponse(e, status=400)
```

Comment:
Create an issue for it please.

Reply:
> **Handle potential `KeyError` when retrieving the tenant ID.**
> 
> Before calling `create_unique_order_id`, add a check to ensure that `request.user.tenant.id` exists to avoid a potential `KeyError`. If the tenant ID doesn't exist, handle the error gracefully and provide an informative error message to the user.

Please create an issue for it